dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/resolver/CategoryOptionGroupResolver.java

```diff
@@ -28,154 +28,95 @@
     }
 
-    private Set<String> resolveCoCFromCog( String categoryOptionGroupUid )
-    {
-        return categoryOptionComboStore.getCategoryOptionCombosByGroupUid( categoryOptionGroupUid ).stream()
-            .map( BaseIdentifiableObject::getUid ).collect( Collectors.toSet() );
-    }
-
-    /**
-     * Resolves a Data Element Operand expression containing one or two Category
-     * Option Group UID to an equivalent expression where the associated
-     * Category Option Combos for the given Category Option Group are "exploded"
-     * into the expression.
-     *
-     * Resolves one of the expressions below:
-     *
-     * 1) #{DEUID.COGUID.AOCUID} 2) #{DEUID.COCUID.COGUID} 3)
-     * #{DEUID.COG1UID.COG2UID}
-     *
-     * to:
-     *
-     * 1) #{DEUID.COCUID1.AOCUID} + #{DEUID.COCUID2.AOCUID} +
-     * #{DEUID.COCUID3.AOCUID} where COCUID1,2,3... are resolved by fetching all
-     * COCUID by COGUID
-     *
-     * 2) #{DEUID.COCUID} + #{DEUID.COCUID1} + #{DEUID.COCUID2} +
-     * #{DEUID.COCUID3} + #{DEUID.COCUID4} where COCUID1,2,3... are resolved by
-     * fetching all COCUID by COGUID
-     *
-     * 3) #{DEUID.COCUID1} + #{DEUID.COCUID2} + #{DEUID.COCUID3} +
-     * #{DEUID.COCUID4} where COCUID1 and COCUID2 are resolved from COG1UID and
-     * COCUID3 and COCUID4 are resolved from COGUID2
-     *
-     * DEUID = Data Element UID COCUID = Category Option Combo UID COGUID =
-     * Category Option Group UID
-     *
-     * @param expression a Data Element Expression
-     * @return an expression containing additional CategoryOptionCombos based on
-     *         the given Category Option Group
-     */
     @Override
-    @Transactional( readOnly = true )
     public String resolve( String expression )
     {
-        // Get a DimensionalItemId from the expression. The expression is parsed
-        // and
-        // each element placed in the DimensionalItemId
         Set<DimensionalItemId> dimItemIds = expressionService.getExpressionDimensionalItemIds( expression,
             INDICATOR_EXPRESSION );
-        List<String> resolvedOperands = new ArrayList<>();
-        if ( isDataElementOperand( dimItemIds ) )
-        {
-            DimensionalItemId dimensionalItemId = dimItemIds.stream().findFirst().get();
-            // First element is always the Data Element Id
-            String dataElementUid = dimensionalItemId.getId0();
-
-            resolvedOperands
-                .addAll( evaluate( dataElementUid, dimensionalItemId.getId1(), dimensionalItemId.getId2() ) );
 
-            resolvedOperands.addAll( evaluate( dataElementUid, dimensionalItemId.getId2(), null ) );
-        }
-        if ( resolvedOperands.isEmpty() )
+        for ( DimensionalItemId id : dimItemIds )
         {
-            // nothing to resolve, add the expression as it is
-            resolvedOperands.add( expression );
+            if ( id.getItem() != null && id.getId1() != null && id.getId1().contains( CATEGORY_OPTION_GROUP_PREFIX ) )
```

Comment:
Replace .contains with .startsWith (clearer intention and marginally faster).

Reply:
done in all resolvers